src/ophyd_async/core/device_save_loader.py

```diff
@@ -0,0 +1,175 @@
+from enum import Enum
+from typing import Any, Dict, Generator, List, Optional, Sequence, Union
+
+import numpy as np
+import numpy.typing as npt
+import yaml
+from bluesky.plan_stubs import abs_set, wait
+from bluesky.utils import Msg
+
+from .device import Device
+from .signal import SignalRW
+
+
+def ndarray_representer(dumper: yaml.Dumper, array: npt.NDArray[Any]) -> yaml.Node:
+    return dumper.represent_sequence("tag:yaml.org,2002:seq", array.tolist())
+
+
+class OphydDumper(yaml.Dumper):
+    def represent_data(self, data: Any) -> Any:
+        if isinstance(data, Enum):
+            return self.represent_data(data.value)
+        return super(OphydDumper, self).represent_data(data)
+
+
+def get_signal_values(
+    signals: Dict[str, SignalRW[Any]], ignore: Optional[List[str]] = None
+) -> Union[Generator[Dict[str, Any], None, None], Dict[str, Any]]:
```

Comment:
I think this is always a generator. @callumforrester is this right for a plan?

```suggestion
) -> Generator[Msg, Sequence[Location], Dict[str, Any]]:
```